Plugins/NativeCamera/Source/Apple/CameraDevice.mm

```diff
@@ -423,7 +424,12 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
 
     void CameraDevice::UpdateCameraTexture(bgfx::TextureHandle textureHandle)
     {
-        arcana::make_task(m_impl->deviceContext->BeforeRenderScheduler(), arcana::cancellation::none(), [this, textureHandle] {
+        arcana::make_task(m_impl->deviceContext->BeforeRenderScheduler(), *m_impl->cancellationSource, [this, textureHandle, cancellationSource{m_impl->cancellationSource}] {
+            if (cancellationSource->cancelled()) {
+                // The stream was closed while we were waiting to render.
+                return;
+            }
```

Comment:
Do we need this? @bghgary you looked at this part of the arcana code recently... do you remember if the cancellation token state is checked before invoking the make_task callback *on the actual dispatcher thread*?

Reply:
I put a breakpoint inside the conditional and I couldn't get it to hit. So I'm inclined to believe it's indeed not required to check for cancelation at the beginning of the async work... but I also don't have a reliable way to ensure I'm actually reproducing the scenario that leads to it being canceled while waiting to render.

Reply:
I'm pretty sure this is already checked in the task continuation code. You should be able to see the check in the call stack if you put a breakpoint at the beginning of the continuation.

Reply:
Arcana does indeed check the cancellation status internally. I've removed my own check, but I've left the capture for the cancellationToken to ensure it's kept alive if the host object is destroyed while the async work is still pending.

Reply:
Yes, this is a stopgap until my other changes set the coding pattern.